### ISSUE TITLE
publish-plugin tasks dependency warnings? 

### ISSUE THREAD

**Issue by nefilim** [gradle/gradle#19331]
### Expected Behavior
The plugin should work out of the box without warnings 

### Current Behavior
```
  - Gradle detected a problem with the following location: 'my-plugin-0.0.4-javadoc.jar'. Reason: Task ':generateMetadataFileForPluginMavenPublication' uses this output of task ':publishPluginJavaDocsJar' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.3.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'my-plugin-0.0.4-sources.jar'. Reason: Task ':generateMetadataFileForPluginMavenPublication' uses this output of task ':publishPluginJar' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.3.1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
```

### Context
https://github.com/nefilim/gradle-semver-plugin/blob/main/build.gradle.kts

attempting to do something like this:

```
tasks.named("generateMetadataFileForPluginMavenPublication") {
    dependsOn("publishPluginJar")
    dependsOn("publishPluginJavaDocsJar")
}
```
fails with tasks not found. these tasks all appear to be internal to the publish plugin, so one would assume the plugin would setup the dependencies correctly? 

### Steps to Reproduce 
run task `publishPlugins`

### Your Environment

Build scan URL: https://scans.gradle.com/s/2p344bf76jjrs

**Comment by eirnym** [comment]
I specify all artifacts in the same publishing.

**Comment by jbartok** [comment]
@nefilim I can confirm the problem you are having. The good news is that it's already fixed in version 1.0 of the Plugin Publish Plugin, not published yet, but hopefully coming soon.

In the meantime the workaround you mentioned will work, you just need to put it inside the `afterEvaluate` block you have in your buildfile.
